### PR TITLE
Lazy attributes

### PR DESCRIPTION
**Pull request by thomasjthomasj**
- [x] Is this Pull Request ready for review?
- [x] Do the tests pass?
- [x] Have the docs been updated?
- [ ] Have any new settings been added?  Have they got comments?  Have they been added to the settings documentation page?

**What does this Pull Request do?**
Moves the 'lazy attribute' logic over from Mcclane to GN Django for use in other libraries. This feature allows models to inherit attributes from parent models and display them as one dictionary.

The original code in Mcclane can be found here: https://github.com/gamernetwork/mcclane/blob/develop/mcclane/core/models.py#L34

**Are there new tests?**
Yes, there's some basic tests to ensure that the correct values are being returned by the models. There are also more 'real world' tests in Mcclane from when the original feature was added.

**What are the relevant Github Issues or Zendesk tickets?**
https://github.com/gamernetwork/gn-django/issues/62

### CODE REVIEW

File: gn_django/models.py [hunk 1]
```diff
@@ -1,10 +1,142 @@
...
+        """
+        attributes = {}
+
+        if hasattr(self, 'override_attributes'):
+            attributes = self.override_attributes
+
+        # Top level models will not have a parent
+        if self.parent:
+            attributes = {**self.parent.attributes, **attributes}
+
```
Comment (johnpooch) [inline gn_django/models.py]: A comment above this `as_json` would be helpful I think.

File: gn_django/models.py [hunk 2]
```diff
@@ -1,10 +1,142 @@
...
+        if as_json:
+            attributes = json.dumps(attributes, indent=True)
+        return attributes
+
+    def get_inherited_attributes(self, as_json=False):
+        """
+        Get all attributes which belong to parents, without attributes specific to this object.
```
Comment (johnpooch) [inline gn_django/models.py]: Mildly triggered by docstrings over 80 chars 😿

Comment (thomasjthomasj) [reply gn_django/models.py]: https://www.youtube.com/watch?gl=BE&v=3BxfpbyV-uc

File: gn_django/models.py (hunk 1 above)
Comment (thomasjthomasj) [reply gn_django/models.py]: Really? I'm not sure how that could be more clear?

**Comment by thomasjthomasj** [comment]
test this please

**Pull request merged**